gulpfile.js

```diff
@@ -1,147 +1,74 @@
-});
-
-
-/******************************************************************************
- * sass
- * Convert Sass files to a single bundled CSS file. Uses auto-prefixer
- * to automatically add required vendor prefixes when needed.
- ******************************************************************************/
-gulp.task('sass', function() {
-    var autoprefixerOpts = {
-        browsers: [
-            'last 2 versions',
-            'iOS >= 7',
-            'Android >= 4',
-            'Explorer >= 10',
-            'ExplorerMobile >= 11'
-        ],
-        cascade: false
-    };
-
-    return gulp.src('app/theme/app.+(ios|md).scss')
-        .pipe(sass({
-            includePaths: [
-                IONIC_DIR,
-                'node_modules/ionicons/dist/scss'
-            ]
-        }))
-        .on('error', function(err) {
-            console.error(err.message);
-            this.emit('end');
-        })
-        .pipe(autoprefixer(autoprefixerOpts))
-        .pipe(gulp.dest('www/build/css'))
-});
-
-
-/******************************************************************************
- * copy.fonts
- * Copy Ionic font files to build directory.
- ******************************************************************************/
-gulp.task('copy.fonts', function() {
-    return gulp.src(IONIC_DIR + 'fonts/**/*.+(ttf|woff|woff2)')
-        .pipe(gulp.dest('www/build/fonts'));
+    gulpWatch = require('gulp-watch'),
+    del = require('del'),
+    runSequence = require('run-sequence'),
+    argv = process.argv;
+
+
+/**
+ * Ionic hooks
+ * Add ':before' or ':after' to any Ionic project command name to run the specified
+ * tasks before or after the command.
+ */
+gulp.task('serve:before', ['watch']);
+gulp.task('emulate:before', ['build']);
+gulp.task('deploy:before', ['build']);
+gulp.task('build:before', ['build']);
+
+// we want to 'watch' when livereloading
+var shouldWatch = argv.indexOf('-l') > -1 || argv.indexOf('--livereload') > -1;
+gulp.task('run:before', [shouldWatch ? 'watch' : 'build']);
+
+/**
+ * Ionic Gulp tasks, for more information on each see
+ * https://github.com/driftyco/ionic-gulp-tasks
+ *
+ * Using these will allow you to stay up to date if the default Ionic 2 build
+ * changes, but you are of course welcome (and encouraged) to customize your
+ * build however you see fit.
+ */
+var buildBrowserify = require('ionic-gulp-browserify-typescript');
+var buildSass = require('ionic-gulp-sass-build');
+var copyHTML = require('ionic-gulp-html-copy');
+var copyFonts = require('ionic-gulp-fonts-copy');
+var copyScripts = require('ionic-gulp-scripts-copy');
+var tslint = require('ionic-gulp-tslint');
+
+var isRelease = argv.indexOf('--release') > -1;
+
+gulp.task('watch', ['clean'], function(done){
+  runSequence(
+    ['sass', 'html', 'fonts', 'scripts'],
+    function(){
+      gulpWatch('app/**/*.scss', function(){ gulp.start('sass'); });
+      gulpWatch('app/**/*.html', function(){ gulp.start('html'); });
```

Comment:
in my local version I had to add *.ts as well so that it would watch changes to the ts files. How do changes in ts files trigger a new build?